File: study/lua/src/app/fight/states/StateShow.lua
--战斗前展示
local State = require("app.fight.states.State")
local StateShow = class("StateShow", State)

local Path = require("app.utils.Path")
local FightConfig = require("app.fight.Config")
local BuffManager = require("app.fight.BuffManager")

--
function StateShow:ctor(entity, skillPlay, skillType, prePosition)
	StateShow.super.ctor(self, entity)
    self._showTime = 0
    self._skillPlay = skillPlay
    self._skillType = skillType
    self._prePosition = prePosition
    self._bShowName = false
end

--播放声音
function StateShow:_playSkillVoice()
    if self._skillPlay and self._skillPlay.battle_voice ~= "0" then
        local mp3 = Path.getSkillVoice(self._skillPlay.battle_voice)
        G_AudioManager:playSound(mp3)
    end
end

--
function StateShow:start()
	StateShow.super.start(self)
    -- if self._entity.partner
    if self._skillType == 2 then
        if self._skillPlay.txt == 0 then
            self:onFinish()
            return
        end
        self:_playSkillVoice()
        if self._skillPlay and self._skillPlay.txt ~= 0 then
            self._entity:showSkill(self._skillPlay.txt)
        end
    elseif self._skillType == 3 then
        local partner = self._entity:getPartner()
        if partner then
            partner:startCombineVice(self._skillPlay, self._prePosition)
            -- self._entity:playDuang(handler(self, self._playSkillVoice))
            self:_playSkillVoice()
        end
        self._entity:playDuang()
        
    end
end

function StateShow:update(f)
    if self._showTime > FightConfig.SHOW_TIME then
        self:onFinish()
    end
    self._showTime = self._showTime + f
end

return StateShow

File: study/ts/script/fight/state/StateShow.ts
import { State } from "./State";
import Entity from "../unit/Entity";
import { FightConfig } from "../FightConfig";
import { Path } from "../../utils/Path";
import { G_AudioManager } from "../../init";
import UnitHero from "../unit/UnitHero";

// 战斗前展示
export class StateShow extends State {
    private _showTime: number;
    private _skillPlay;
    private _skillType;
    private _prePosition;

    constructor(entity: Entity, skillPlay, skillType, prePosition) {
        super(entity);
        this.cName = "StateShow";
        this._showTime = 0;
        this._skillPlay = skillPlay;
        this._skillType = skillType;
        this._prePosition = prePosition;
        this.bShowName = false;
    }

    public start() {
        super.start();
        if (this._skillType == 2) {
            if (this._skillPlay.txt == 0) {
                this.onFinish();
            }
            this.playSkillVoice();
            if (this._skillPlay && this._skillPlay.txt != 0) {
                (this.entity as UnitHero).showSkill(this._skillPlay.txt);
            }
        }
        else if (this._skillType == 3) {
            let partner = this.entity.getPartner();
            if (partner) {
                partner.startCombineVice(this._skillPlay, this._prePosition)
                this.playSkillVoice();
            }
            (this.entity as UnitHero).playDuang();
        }
    }

    public update(f: number) {
        if (this._showTime > FightConfig.SHOW_TIME) {
            this.onFinish();
        }
        this._showTime += f;
    }

    private playSkillVoice() {
        if (this._skillPlay && this._skillPlay.battle_voice != 0) {
            let mp3 = Path.getSkillVoice(this._skillPlay.battle_voice);
            G_AudioManager.playSound(mp3);
        }
    }
}